src/gc/transaction_level_gc_manager.cpp

```diff
@@ -262,77 +371,83 @@ void TransactionLevelGCManager::AddToRecycleMap(
     PELOTON_ASSERT(database != nullptr);
     if (table_oid == INVALID_OID) {
       storage_manager->RemoveDatabaseFromStorageManager(database_oid);
+      LOG_TRACE("GCing database %u", database_oid);
       continue;
     }
     auto table = database->GetTableWithOid(table_oid);
     PELOTON_ASSERT(table != nullptr);
     if (index_oid == INVALID_OID) {
       database->DropTableWithOid(table_oid);
-      LOG_DEBUG("GCing table %u", table_oid);
+      LOG_TRACE("GCing table %u", table_oid);
       continue;
     }
     auto index = table->GetIndexWithOid(index_oid);
     PELOTON_ASSERT(index != nullptr);
     table->DropIndexWithOid(index_oid);
-    LOG_DEBUG("GCing index %u", index_oid);
+    LOG_TRACE("GCing index %u", index_oid);
   }
 
   delete txn_ctx;
 }
 
-// this function returns a free tuple slot, if one exists
-// called by data_table.
-ItemPointer TransactionLevelGCManager::ReturnFreeSlot(const oid_t &table_id) {
-  // for catalog tables, we directly return invalid item pointer.
-  if (recycle_queue_map_.find(table_id) == recycle_queue_map_.end()) {
+// looks for a free tuple slot that can now be reused
+// called by data_table, which passes in a pointer to itself
+ItemPointer TransactionLevelGCManager::GetRecycledTupleSlot(
+    storage::DataTable *table) {
+  auto recycle_queue = GetTableRecycleQueue(table->GetOid());
```

Comment:
We had changed it since the TG compaction stuff needed the table pointer, but with that stuff stripped out it makes more sense to change it back to the oid being passed in. Good catch.